src/main/java/dev/galacticraft/mod/client/gui/screen/ingame/ConfirmFlagScreen.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019-2024 Team Galacticraft
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package dev.galacticraft.mod.client.gui.screen.ingame;
+
+import it.unimi.dsi.fastutil.booleans.BooleanConsumer;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.screens.ConfirmScreen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.Mth;
+
+public class ConfirmFlagScreen extends ConfirmScreen {
+    protected final ResourceLocation imageLocation;
+    public ConfirmFlagScreen(BooleanConsumer booleanConsumer, ResourceLocation flagImage, Component component, Component component2) {
+        super(booleanConsumer, component, component2);
+        this.imageLocation = flagImage;
+    }
+
+    private int titleTop() {
+        int i = this.height/2;
+        return Mth.clamp(i - 20 - this.font.lineHeight, 10, 80);
+    }
+
+    private int flagTop() {
+        return this.titleTop() + 20;
+    }
+
+    @Override
+    public void render(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+        super.render(graphics, mouseX, mouseY, delta);
+        graphics.pose().pushPose();
+        graphics.pose().translate((double) this.width/2 - 24, this.flagTop(), 0.0);
+        //graphics.pose().scale(1.5f, 1.5f, 1.5f);
+        graphics.blit(this.imageLocation, 0, 0, 0, 0, 0, 48, 32, 48, 32);
+        graphics.pose().popPose();
```

Comment:
Could the translation be moved into the blit instead of modifying the pose stack?

Reply:
See other comment; I was trying out scaling the flag on this screen too, so I used the same method. I can change it back to just passing the position to graphics.blit().